src/plugins/global-settings/preset-controls/index.php

```diff
@@ -0,0 +1,172 @@
+<?php
+/**
+ * Size and Spacing Preset Controls
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'Stackable_Size_And_Spacing_Preset_Controls' ) ) {
+
+	/**
+	 * Size and Spacing Preset Controls
+	 */
+    class Stackable_Size_And_Spacing_Preset_Controls {
+		public const PRESET_MAPPING = array(
+			'fontSizes' => array(
+				'settings' => array('typography', 'fontSizes' ),
+				'prefix' => 'font-size',
+			),
+			'spacingSizes' => array(
+				'settings' => array( 'spacing', 'spacingSizes' ),
+				'prefix' => 'spacing',
+			),
+			'blockHeights' => array(
+				'settings' => array( 'blockHeights' ),
+				'prefix' => 'block-height',
+			),
+			'borderRadius' => array(
+				'settings' => array( 'borderRadius' ),
+				'prefix' => 'border-radius',
+			),
+		);
+
+		public $custom_presets;
+		public $theme_presets;
+		public $default_presets;
+		public $stackable_presets;
+
+		/**
+		 * Initialize
+		 */
+  		function __construct() {
+			$this->custom_presets = get_option( 'stackable_global_custom_preset_controls' );
+			$this->theme_presets = WP_Theme_JSON_Resolver::get_theme_data()->get_settings();
+			$this->default_presets = WP_Theme_JSON_Resolver::get_core_data()->get_settings();
+			$this->stackable_presets = $this->load_presets( __DIR__ . '/presets.json');
+
+			add_filter( 'stackable_inline_styles_nodep', array( $this, 'add_preset_controls_styles' ) );
+			add_filter( 'stackable_inline_editor_styles', array( $this, 'add_preset_controls_styles' ) );
```

Comment:
These will be called even in the front end where I don't think the presets are needed.

I recommend to place the code the populates `custom_presets`, `theme_presets` etc into a new function:

```php
public function load_presets() {
    $this->custom_presets = get_option( 'stackable_global_custom_preset_controls' );
    // ...
}
```

Then calling `$this->load_presets()` when the values are about to be used.

This is so that the loading process will only be called when needed (currently this performed for every page load)